pydarn/plotting/maps.py

```diff
@@ -1030,3 +1030,117 @@ def plot_potential_contours(cls, fit_coefficient: list, lat_min: list,
                     color=pot_minmax_color, zorder=5.0)
         plt.scatter(np.radians(min_mlon), min_mlat, marker='_', s=70,
                     color=pot_minmax_color, zorder=5.0)
+        
+    
+
+    class TimeSeriesParams(Enum):
+        """
+        Enum class to hold the parameters that can be plotted
+        Members
+        ------------
+        NUM_VECTORS:
+        the number of aviliable vectors in the map
+        IMF_BY:
+        The magnitude of the IMF By component in nT
+        IMF_BZ:
+        The magnitude of the IMF Bz component in nT
+        KP:
+        The Kp index
+        HMB_MAX:
+        The farthest equatorward extent of the HMB
+
+        """
+        NUM_VECTORS = 'num_vectors'
+        IMF_BY = 'imf_by'
+        IMF_BZ = 'imf_bz'
+        KP = 'kp'
+        HMB_MAX = 'hmb_max'     
+        
+    @classmethod
+    def plot_time_series(cls, dmap_data: List[dict],
+                     parameter: Enum = TimeSeriesParams.NUM_VECTORS,
+                     start_record: int = 0,end_record: int = 1, start_time: dt.datetime = None,
+                     end_time: dt.datetime = None):
+        '''
+        Plot time series of various map parameters
+        
+        Params
+        ----------
+        dmap_data: List[dict]
+            List of dictionaries containing the data to be plotted
+        start_record: int
+            time index that we will plot from
+        end_record: int
+            time index that we will plot to (-1 will plot the entire period)
+        start_time: dt.datetime
+            Start time of the data
+        end_time: dt.datetime
+            End time of the data
+            '''
+        #create a new matplotlib figure
+        fig = plt.figure()
+        #if end_record is -1, set it to the last record
+        if end_record == -1:
+            end_record = len(dmap_data)-1
+        #determine the start and end record
+        if start_time is not None and end_time is not None:
+            raise NotImplementedError('use of datetime objects is not yet implemeneted')
```

Comment:
So I removed this and uncommented the two lines below and I can get find_record to work if I use:

```python
file_path = "/Users/carley/Documents/data/maps/20190421.s.map2"
data = pydarn.SuperDARNRead().read_dmap(file_path)
start_time = dt.datetime(2019,4,21,11,0)
end_time = dt.datetime(2019,4,21,13,0)
pydarn.Maps.plot_time_series(data, parameter=pydarn.Maps.TimeSeriesParams.HMB_MAX,
                             start_time=start_time, end_time=end_time)
plt.show()
```

I think there was an old issue that didn't allow you to put seconds in there, because it would break? But either way it works for me using these date times. What is the error you got, we can maybe do a find_map_record or something? (or add an option in find_records to find nearest second/minute/2minutes then it's useful for all data types, this is probably a different issue to fix+PR though? Can talk about it friday at the meeting)

As map files have data every two mins, if I use any other number for seconds than 0 I get:
```
Traceback (most recent call last):
  File "/Users/carley/Documents/testing_code/venv-timemap/pydarn/../../test_maptimeseries.py", line 9, in <module>
    pydarn.Maps.plot_time_series(data, parameter=pydarn.Maps.TimeSeriesParams.HMB_MAX,
  File "/Users/carley/Documents/testing_code/venv-timemap/lib/python3.10/site-packages/pydarn/plotting/maps.py", line 1088, in plot_time_series
    start_record = find_record(dmap_data, start_time) #TODO: Fix the error that gets thrown when using this
  File "/Users/carley/Documents/testing_code/venv-timemap/lib/python3.10/site-packages/pydarn/utils/plotting.py", line 67, in find_record
    raise plot_exceptions.NoDataFoundError(parameter, start_time=start_time)
NameError: name 'parameter' is not defined
```
Which I guess is expected because we don't read in any parameter to the function.... so this an error in the exception it raises, and a different problem to solve. but essentially, it's looking for data at a time where data doesn't exist. I think the solution here is to make a new find_map_record that checks nearest minute, rather than nearest second?